File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingGradeForm.ts
import { BooleanEditor, initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface TrainingGradeForm {
    Grade: StringEditor;
    TrainerName: StringEditor;
    TrainerNumber: StringEditor;
    ReferenceNumber: StringEditor;
    ScheduleType: StringEditor;
    CourseName: StringEditor;
    CourseCode: StringEditor;
    Department: StringEditor;
    Specialization: StringEditor;
    TrainingLevel: StringEditor;
    TrainingTerm: StringEditor;
    IsActive: BooleanEditor;
}

export class TrainingGradeForm extends PrefixedContext {
    static readonly formKey = 'Administration.TrainingGrade';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TrainingGradeForm.init) {
            TrainingGradeForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(TrainingGradeForm, [
                'Grade', w0,
                'TrainerName', w0,
                'TrainerNumber', w0,
                'ReferenceNumber', w0,
                'ScheduleType', w0,
                'CourseName', w0,
                'CourseCode', w0,
                'Department', w0,
                'Specialization', w0,
                'TrainingLevel', w0,
                'TrainingTerm', w0,
                'IsActive', w1
            ]);
        }
    }
}

File: QadeerApp.Web/Modules/Administration/TrainingGrade/TrainingGradeForm.cs
namespace QadeerApp.Administration.Forms;

[FormScript("Administration.TrainingGrade")]
[BasedOnRow(typeof(TrainingGradeRow), CheckNames = true)]
public class TrainingGradeForm
{
    public string Grade { get; set; }
    public string TrainerName { get; set; }
    public string TrainerNumber { get; set; }
    public string ReferenceNumber { get; set; }
    public string ScheduleType { get; set; }
    public string CourseName { get; set; }
    public string CourseCode { get; set; }
    public string Department { get; set; }
    public string Specialization { get; set; }
    public string TrainingLevel { get; set; }
    public string TrainingTerm { get; set; }
    public bool IsActive { get; set; }
}
